### ISSUE TITLE
Button with a Command get disabled when selecting an item in DataGrid and then invoke ChangeTheme

### ISSUE THREAD

**Issue by UlyssesWu** [MahApps/MahApps.Metro#4224]
## Describe the bug

``` xaml
    <Window.Resources>
        <RoutedUICommand x:Key="QuickLoadCommand" Text="Quick Load" />
    </Window.Resources>
    <Window.CommandBindings>
        <CommandBinding Command="{StaticResource QuickLoadCommand}"
                        Executed="QuickLoad"/>
    </Window.CommandBindings>

<Button Command="{StaticResource QuickLoadCommand}" Content="Test"></Button>
<mah:ToggleSwitch x:Name="ThemeToggle" Toggled="ChangeTheme" />
```

The button is enabled by default.
![image](https://user-images.githubusercontent.com/4312781/144749948-8c4cc719-fe21-4616-9f94-9d05ae99ac4a.png)

But if you select a row in DataGrid and then call `ThemeManager.Current.ChangeTheme`:
![image](https://user-images.githubusercontent.com/4312781/144750000-c8ad2339-3526-42e4-a1c0-f20c7701a231.png)

The button is disabled now, and it returns to be enabled when you click in DataGrid.

## Steps to reproduce

1. clone  https://github.com/UlyssesWu/MahApps.Issue.ButtonWithCommand 
2. nuget restore, then Run MahAppsTest
3. Confirm the `Test` Button is enabled. If you click it, the Window's title is set to current time
4. Click the first row in DataGrid
5. Click the toggle button, this will change theme to dark mode
6. Check the `Test` Button

## Expected behavior

The `Test` button is enabled. If you click it, the Window's title is set to current time

## Actual behavior

The `Test` button is disabled, and it returns to be enabled when you click a row in DataGrid.

## Environment

```
MahApps.Metro version: v2.4.9
Windows build number: Win10 21H2
Visual Studio: 2022 17.0.1
Target Framework: .NET Framework 4.8
```

**Comment by batzen** [comment]
Could you try adding `CommandManager.InvalidateRequerySuggested();` after switching the theme?

**Comment by UlyssesWu** [comment]
Tried, nothing changes.

**Comment by punker76** [comment]
@UlyssesWu After some trial and errors and some googling I found a solution, but I don't understand why this is an issue if I use MahApps in this scenario. I tried different ways to understand what's happening:

1. try without DataGrid MahApps Style => it works
2. try with a button outside the Toolbar => it works
3. try without RoutedUICommand => it works

So at the end it seems like a special thing with the styles of MahApps and something else...

My solution for you is now to set the CommandTarget for the Button to the main window: 

```xaml
<Button CommandTarget="{Binding RelativeSource={RelativeSource FindAncestor, AncestorType={x:Type Window}}}" Command="{StaticResource QuickLoadCommand}" Content="Test"></Button>
```

With this the Button works as expected.

Here are the links

- https://wpf.2000things.com/2014/06/10/1090-sender-vs-source-in-commandbinding-event-handlers/
- https://wpf.2000things.com/tag/commandtarget/
